# TODO:
- Exercises:
	- [ ] Parkinson GWAS and UCSC
	- [ ] Influenza virus trains (pw alignment)
	- [ ] Protein alignment and virulence factors
	- [ ] Horizontal gene transfer (blast)
	- [ ] Archaea in the tree of life (clustering)
	- [ ] Evolution of the Whale (phylogenetics)
	- [ ] Gene finding (ORF on plasmid and genscan)
	- [ ] RNA folding
	- [ ] eDNA barcoding?
- Programming projects:
	- [ ] Put projects in individual PDFs
	- [ ] Adjust programming projects to fit progression
	- [ ] Pam project (with matrix multiplication)
	- [ ] eDNA BioPython project or Robinson-Foulds distance (based on old exam set) 
	- [ ] Primer project from old exam set
- Curriculum:
	- [ ] 


# Masterplan

- Ny kursusplan
- Ny eksamensform:
    1. programmeringsopgaver
    2. multiplechoice
    3. noget med nogen praktiske øvelser/beregninger på laptop med software de har installeret til øvelserne (data skal så downloades via Digital Eksamen)
    4. En enkelt eller to fritekst opgaver: """ Blast er ... """ (jeg kan lave et script er printer hver studerendes svar pænt og prompter for en karakte som så bliver skrevrt ned i en fil til sidst.)
- Afskriv/tilpas bioinformatikøvelser
- Update programmeringsnoter
- Evt. nye projekter (PAM and Robinson-Foulds)
- Nyt pensum: kompendium af reviews / NCBI resources / lette artikler / Durbin udklip / Expl. Bioinf. udklip
- Nye kapitler i noter hvor jeg ikke kan finde noget godt. Evt. bare lidt supplerende/støttende tekst til at starte med.


# New course plan

- Et syv ugers kursus giver ikke nok routine til at færdighederne hænger ved. Abstrakt tænkning tager tid og øvelse at indarbejde.
- Uge 5,6,7: Øvelser/routine i programmering: ikke hensigtsmæssigt som forelæsninger, men der er ikke andet at flyde i de forelæsninger fordi jeg ikke kan begynde på bioinformatik
- Uge 8-14: for få forelæsninger til meget bioinformatik stof: Det kommer til at gå lidt for hurtigt for mange studerende. Kunne med fordel bredes ud ved at inkludere bioinformatik i uge 5-7.
- Med bioinformatik bredt ud over flere uger, vil øvelser bedre kunne integrere programmering og praktisk anvendelse af bioinformatik værktøjer, f.eks. BioPython. Dvs. mere tid flere praktiske færdigheder. Studerende der er klar til at anvende færdigheder i projekter og specialer. 
- Et omstrukturert kursus ville kunne tilgodese relevante færdigheder for både rigtige biologer og molekylærbiologer. Kurset er i forvejen begrænset til DNA bioinformatik og man kunne tilføje øvelser om genetisk diversitet.



# Lectures

- Try to lecture concretely on how to build code from the individual building blocks. How you stay on top of things by for each line of code added.
- The summing up and evolution lecture, where I ask them to remember all the building blocks does not work so well. There needs to be a lot of pentameter interaction for it to work.
- The Guessing game lecture does not work well. It is too hard to make it an interaction with the students. Maybe just make it a small thing, in another lecture that as an example of how to use the input function. I could just show them the game I made and have them download and try it.
- Shorten assembly lecture. there was no time for exercises. 
- Cut down global alignment recap. Otherwise there is not enough time to explain affine gap cost. Affine gap cost should also be explained to show what it corresponds to in the animation explaining alignment with linear cost.
- Maybe split main HMM lecture and only do Viterbi in the first one (with some applications) and then do posterior decoding in another lecture (e.g. when you do TMHMM).
- Clustering lecture can be made to fill two lectures (with a small project introduction at the end.) As it is the lecture takes an hour and 15, at a relative high pace.
- Maybe use Laurits Archaic HMM as HMM example (simple two state with higher probability of emitting "difference" in archaic.)
- Make a Menti poll every week about workload, exercises, etc.

# Course plan
- Maybe do HMMs before multiple alignments (like in the book) so I can talk about profile HMMs and aligning profiles before I talk about progressive alignment.

# Programming projects

add some extra tests to assembly project so students cannot just return the requested value. 

Rewrite programming project so the amount of hits fit with the sequence of projects. E.g. HIV project ends with "You are now a programmer" which is a bit lame if it is the third project.

# Exercises
Think about including theoretical math exercises to train familiarity with formulas and algorithms (like in the Durbin book).