## Notes for chapter 5 {-}

For a description of the web exercise see the attached PDF file, which is from Exploring Bioinformatics: A Project-Based Approach 2nd, Caroline St. Clair; Jonathan Visick, Jones & Bartlett Learning, 2013, ISBN: 978-1-284-03424-0. A few things have changed since the book was published: 

In part two of the exercise (More tools for Exploring Protein Function), the Pfam and MOTIF online tools work best if you paste the FASTA sequence directly into the input fields. As an alternative to the DAS tool (which is a little hard to interpret), you can use [TMHMM](http://www.cbs.dtu.dk/services/TMHMM/), which is a hidden Markov model for predicting transmembrane helices.

Another tool you can use for Human genes and the cellular location of their gene products Gene Cards https://www.genecards.org

Yet another good resource is [Uniprot](https://www.uniprot.org), which is a comprehensive platform for protein sequence and annotations data arising from a collaboration between EMBL-EBI and the swiss institute of bioinformatics. You can access anything from disease involvement of proteins, their cellular location, and taxonomy. To use it to find the cellular location of a protein, retrieve the GenBank accession number (The id after "GenBank:" at the top of the GenBank entry) and use it in the Uniprot search bar. If you scroll down the result page, you will get info about cellular location.






